src/libraries/System.Text.Json/gen/JsonSourceGenerator.Roslyn4.0.cs

```diff
@@ -39,11 +36,11 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                     Parser.JsonSerializableAttributeFullName,
                     (node, _) => node is ClassDeclarationSyntax,
                     (context, _) => (ContextClass: (ClassDeclarationSyntax)context.TargetNode, context.SemanticModel))
-                .Combine(knownTypeSymbols)
                 .Select(static (tuple, cancellationToken) =>
                 {
-                    Parser parser = new(tuple.Right);
-                    ContextGenerationSpec? contextGenerationSpec = parser.ParseContextGenerationSpec(tuple.Left.ContextClass, tuple.Left.SemanticModel, cancellationToken);
+                    KnownTypeSymbols knownTypes = new(tuple.SemanticModel.Compilation);
```

Comment:
The reason that this has no impact on liveliness is because it's still running on every keystroke. The `transform` callback will be executed on every keystroke, because semantic information can change on any edit. Practically, I don't believe that this change will have a benefit; in fact, it may be a deoptimization since `KnownTypeSymbols` won't be shared between callbacks.

Reply:
Is it not possible though for the transform callback to result in equal values under certain keystrokes? (in this case the `(ClassDeclarationSyntax, SemanticModel)` tuple). Or is it always the case that a keystroke will result in a new SyntaxNode/SemanticModel instance being created?

Reply:
New SyntaxNode, no. New SemanticModel, yes, that will be new every time.

Reply:
Yeah, I suppose that makes sense. Ok, I think we can leave it as-is in that case.

Reply:
note: even if the number of hits is zero, you're still calling into this select every time.  and you'll have to ensrue that 100% of this code is lazy, or risk regressions.  Easiest is to do what @tarekgh did and literally just bail out as the very first thing if hte hit-count is zero.